lib/system.nim

```diff
@@ -906,7 +906,15 @@ proc default*[T](_: typedesc[T]): T {.magic: "Default", noSideEffect.} =
 
 proc reset*[T](obj: var T) {.noSideEffect.} =
   ## Resets an object `obj` to its default value.
-  obj = default(typeof(obj))
+  when nimvm:
+    obj = default(typeof(obj))
+  else:
+    when defined(gcDestructors):
+      {.cast(noSideEffect), cast(raises: []), cast(tags: []).}:
```

Comment:
> A =destroy is implicitly annotated with .raises: []; a destructor should not raise exceptions. For backwards compatibility the compiler produces a warning for a =destroy that does raise.
I read the manual again. It seems that to keep backwards compatibility, I need to suppress raises. As to tags, maybe I should let it bubble up.

Reply:
It's fine this way IMO.